extension/extension.go

```diff
@@ -44,18 +44,44 @@ type PipelineWatcher interface {
 }
 
 // CreateSettings is passed to Factory.Create(...) function.
-type CreateSettings = component.ExtensionCreateSettings //nolint:staticcheck
+type CreateSettings struct {
+	// ID returns the ID of the component that will be created.
+	ID component.ID
+
+	component.TelemetrySettings
+
+	// BuildInfo can be used by components for informational purposes
+	BuildInfo component.BuildInfo
+}
 
 // CreateFunc is the equivalent of Factory.Create(...) function.
 type CreateFunc func(context.Context, CreateSettings, component.Config) (Extension, error)
 
-// CreateExtension implements ExtensionFactory.CreateExtension.
-func (f CreateFunc) CreateExtension(ctx context.Context, set CreateSettings, cfg component.Config) (Extension, error) {
+// Create implements Factory.Create.
+func (f CreateFunc) Create(ctx context.Context, set CreateSettings, cfg component.Config) (Extension, error) {
 	return f(ctx, set, cfg)
 }
 
-// Factory is a factory for extensions to the service.
-type Factory = component.ExtensionFactory //nolint:staticcheck
+// Deprecated: [v0.69.0] use Create.
+func (f CreateFunc) CreateExtension(ctx context.Context, set CreateSettings, cfg component.Config) (Extension, error) {
+	return f.Create(ctx, set, cfg)
+}
+
+type Factory interface {
+	component.Factory
+
+	// Create creates an extension based on the given config.
+	Create(ctx context.Context, set CreateSettings, cfg component.Config) (Extension, error)
```

Comment:
I know this is a good change... But I would split this into a separate PR

Reply:
👍  I removed `Create` and `Stability` methods.